pages/reminders/index.js

```diff
@@ -81,6 +86,54 @@ export default function Reminders() {
     );
   };
 
+  useEffect(() => {
+    async function requestNotificationPermission() {
+      if (isLoadingPublicKey) {
+        return;
+      }
+
+      if (!("Notification" in window)) {
+        return;
+      }
+
+      if (!("serviceWorker" in navigator) || !("PushManager" in window)) {
+        return;
+      }
+
+      if (Notification.permission === "default") {
+        await Notification.requestPermission();
+
+        if (Notification.permission === "granted") {
+          const { publicKey } = publicKeyResponse;
+          const publicKeyArray = urlBase64ToUint8Array(publicKey);
+
+          const serviceWorkerRegistration = await navigator.serviceWorker.ready;
+          const existingSubscription =
+            await serviceWorkerRegistration.pushManager.getSubscription();
+          if (!existingSubscription) {
+            const subscription =
+              await serviceWorkerRegistration.pushManager.subscribe({
+                userVisibleOnly: true,
+                applicationServerKey: publicKeyArray,
+              });
+
+            const response = await fetch("/api/push/subscribe", {
+              method: "POST",
+              headers: {
+                "Content-Type": "application/json",
+              },
+              body: JSON.stringify(subscription),
+            });
+          }
+        } else {
+          console.log("Notifications are blocked by the user.");
+        }
+      }
```

Comment:
by flipping the check in your if statement you can reduce the nesting of if blocks and therefore improve readability:
```js
if (Notification.permission !== "default") {
  return
}

await Notification.requestPermission();

if (Notification.permission !== "granted") {
  console.log("Notifications are blocked by the user.");
  return
}

const { publicKey } = publicKeyResponse;
const publicKeyArray = urlBase64ToUint8Array(publicKey);

const serviceWorkerRegistration = await navigator.serviceWorker.ready;
...

if (existingSubscription) {
  return
}

const subscription =
...
```